lib/tapioca/runtime/reflection.rb

```diff
@@ -19,6 +19,7 @@ module Reflection
       PROTECTED_INSTANCE_METHODS_METHOD = T.let(Module.instance_method(:protected_instance_methods), UnboundMethod)
       PRIVATE_INSTANCE_METHODS_METHOD = T.let(Module.instance_method(:private_instance_methods), UnboundMethod)
       METHOD_METHOD = T.let(Kernel.instance_method(:method), UnboundMethod)
+      UNDEFINED_CONSTANT = T.let(Module.new.freeze, Module)
```

Comment:
What do you think about providing an helper method to use this like `Runtime::Reflection.is_constant_undefined?`?

Reply:
What would the method do? Wouldn't it end up being the same as `constantize`?

Reply:
I tried putting this method together, but it actually fails in Sorbet runtime validations
```ruby
sig { params(constant: BasicObject).returns(T::Boolean) }
def constant_defined?(constant)
  !UNDEFINED_CONSTANT.eql?(constant)
end
```
We have a test in the pipeline spec that verifies we can process `BasicObject` and it fails with
```
NoMethodError: undefined method `is_a?' for #<BasicObjectTest::B:0x00000001184d5cb0>
  obj.is_a?(@raw_type)
                 ^^^^^^
  sorbet-runtime-0.5.10326/lib/types/types/simple.rb:24:in `valid?'
```
Any ideas on how to structure this that wouldn't break the simple validator?

Reply:
Done.